protocol/x/delaymsg/keeper/dispatch.go

```diff
@@ -17,6 +17,13 @@ func DispatchMessagesForBlock(k types.DelayMsgKeeper, ctx sdk.Context) {
 		return
 	}
 
+	// Maintain a list of events emitted by all delayed messages executed in this block.
+	// As each delayed message is executed in a cached context, their emitted events need
```

Comment:
Nit: I think, based on your explanation, that this comment is inaccurate? "As each handler creates collects events in it's own EventManger, the emitted events need to be explicitly added to the current context." or some such.